### PR TITLE
fix header define for shader.

### PR DESCRIPTION
**Pull request by Mee-gu**
fix header define for shader.

### CODE REVIEW

File: cocos/renderer/backend/opengl/ProgramGL.cpp [hunk 1]
```diff
@@ -37,16 +37,16 @@ CC_BACKEND_BEGIN
...
-    std::string headersDef("#ifdef GL_FRAGMENT_PRECISION_HIGH\n" \
-                           "precision highp float;\n precision highp int;\n" \
-                           "#else\n" \
-                           "precision mediump float;\n precision mediump int;\n"
-                           "#endif\n");
-    headersDef.append(fragmentShader);
-    _fragmentShaderModule = static_cast<ShaderModuleGL*>(ShaderCache::newFragmentShaderModule(headersDef));
+    //some device required manually specify the precision qualifiers for vertex shader.
+    std::string vsPreDefine("#version 100\n precision highp float;\n precision highp int;\n");
+    std::string fsPreDefine("precision mediump float;\n precision mediump int;\n");
```
Comment (minggo) [inline cocos/renderer/backend/opengl/ProgramGL.cpp:43]: vsPreDefine and fsPreDefine are const strings, it is better to make them as const member variables.

Comment (Mee-gu) [reply cocos/renderer/backend/opengl/ProgramGL.cpp:43]: I think `vsPreDefine` is better as a non-const string since it will append with shader source.

Comment (minggo) [reply cocos/renderer/backend/opengl/ProgramGL.cpp:43]: But you can insert vsPreDefine at the begin of the shader.

Comment (minggo) [reply cocos/renderer/backend/opengl/ProgramGL.cpp:43]: Ok, it is not suitable, ignore it.

**Pull request merged**